scripts/setup-ssh-keys.sh

```
@@ -0,0 +1,1069 @@
+            while IFS= read -r subline; do
+                [[ -z "${subline// }" ]] && break
+                [[ ! "$subline" =~ ^[[:space:]]+ ]] && break
+                
+                if [[ "$subline" =~ ^[[:space:]]+hostname:[[:space:]]*(.+)$ ]]; then
+                    hostname="${BASH_REMATCH[1]// }"
+                elif [[ "$subline" =~ ^[[:space:]]+user:[[:space:]]*(.+)$ ]]; then
+                    user="${BASH_REMATCH[1]// }"
+                elif [[ "$subline" =~ ^[[:space:]]+port:[[:space:]]*([0-9]+)$ ]]; then
+                    port="${BASH_REMATCH[1]}"
+                fi
+            done <<< "$(tail -n +$(($(grep -n "^[[:space:]]*${host_name}:" "$config_file" | cut -d: -f1) + 1)) "$config_file")"
```

Shellcheck (0.10.0)

[info] Make sure not to read and write the same file in the same pipeline.

(SC2094)